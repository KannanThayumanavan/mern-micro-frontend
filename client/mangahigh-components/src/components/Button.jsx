import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ type, buttonOnClick, buttonLabel }) => (
	<button 
		type="button" 
		className={`btn btn-${type}`}
		onClick={buttonOnClick}
		value={buttonLabel}
	>
		{buttonLabel}
	</button>
);

Button.propTypes = {
	type: PropTypes.string.isRequired, 
	buttonOnClick: PropTypes.func.isRequired,
	buttonLabel: PropTypes.string.isRequired,
};

export default Button;