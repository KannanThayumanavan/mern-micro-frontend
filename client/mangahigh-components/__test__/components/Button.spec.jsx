import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '../../src/components/Button';

describe('Test Button Component', () => {
	it('should render correctly with required props', () => {
		const renderedModule = shallow(<Button
      type='primary'
      buttonLabel='Test Button'
      buttonOnClick={jest.fn()}
    />);
    expect(renderedModule).toMatchSnapshot();
	});
});