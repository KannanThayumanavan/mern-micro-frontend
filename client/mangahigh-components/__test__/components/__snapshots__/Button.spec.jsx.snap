// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component should render correctly with required props 1`] = `
<button
  className="btn btn-primary"
  onClick={[MockFunction]}
  type="button"
  value="Test Button"
>
  Test Button
</button>
`;
